pkg/collector/tcp.go

```diff
@@ -79,31 +82,39 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 				return
 			}
 			klog.V(2).Infof("Receiving %d bytes from %s", size, address)
-			buffBytes := make([]byte, size)
-			copy(buffBytes, buff[:size])
+			klog.V(2).Infof("Remaining bytes from last read length: %v", len(remainingBytes))
+			size = size + len(remainingBytes)
+			copiedBytes := make([]byte, size)
+			copy(copiedBytes[:len(remainingBytes)], remainingBytes)
+			copy(copiedBytes[len(remainingBytes):], buff[:size-len(remainingBytes)])
 			for size > 0 {
-				length, err := getMessageLength(bytes.NewBuffer(buffBytes))
+				bytesBuffer := bytes.NewBuffer(copiedBytes)
+				length, version, err := getMessageLength(bytesBuffer)
 				if err != nil {
 					klog.Error(err)
 					cp.deleteClient(address)
 					return
 				}
 				if size < length {
-					klog.Errorf("Message length %v is larger than size read from buffer %v", length, size)
-					cp.deleteClient(address)
-					return
+					klog.V(2).Infof("Message length %v is larger than size read from buffer %v", length, size)
+					remainingBytes = make([]byte, len(copiedBytes))
+					copy(remainingBytes, copiedBytes)
+					copiedBytes = nil
+					break
 				}
 				size = size - length
-				// get the message here
-				message, err := cp.decodePacket(bytes.NewBuffer(buffBytes[0:length]), address)
+				// Truncate the bytesBuffer here to decode the packet. 4B are already
+				// read as part of getMessageLength.
+				bytesBuffer.Truncate(length - 4)
```

Comment:
`Truncate` removes data "at the end" of the buffer, not at the front

Since you already should have consumed these 4 bytes in `getMessageLength`, I am not sure why this line is needed at all.